### PR TITLE
Add duplicate feedstock as output

### LINKED ISSUE #29: Make noarch: python?
Now that we no longer support python2, we can make this `noarch: python`.

This would then probably collapse the need for https://github.com/conda-forge/antlr4-python3-runtime-feedstock/issues/1

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #29

TODO:
* [x] Inform maintainer of old feedstock
  * [x] Should the old maintainer be invited to co-maintain?
* [x] Add to feedstock-outputs
  * https://github.com/conda-forge/feedstock-outputs/pull/27
* [x] Mark noarch as broken
  * https://github.com/conda-forge/admin-requests/pull/352
  * Once this is merged, a release for v4.9.2 should be made from here so that the removed packages are then available through this feedstock.
  * https://github.com/conda-forge/antlr-python-runtime-feedstock/pull/32
* [x] Add issue to other feedstock explaining duplicate
  * https://github.com/conda-forge/antlr4-python3-runtime-feedstock/issues/3
* [x] Archive other feedstock
* [x] Convert to noarch
  * Currently blocked on https://github.com/antlr/antlr4/issues/3350
* [ ] Close issues and PRs on duplicate feedstock 
  * https://github.com/conda-forge/antlr4-python3-runtime-feedstock/issues/1
  * https://github.com/conda-forge/antlr4-python3-runtime-feedstock/pull/2

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

**Comment by BastianZim** [comment]
@bollwyvl I will add a more detailed description of my plan in a sec but would you mind reviewing this package already? It shouldn't be merged though yet.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {% set name = "antlr4-python3-runtime" %}
+{% set old_name = "antlr-python-runtime" %}
 {% set version = "4.9.3" %}
 
+
 package:
-  name: antlr-python-runtime
+  name: {{ old_name|lower }}-meta
```
Comment (bollwyvl) [inline recipe/meta.yaml]: maybe just be explicit everywhere... none of them are upper-cased anyway, and it gets pretty ugly down in the weeds with lots of template stuff

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,32 +14,48 @@ source:
...
-  imports:
-    - antlr4
-    - antlr4.atn
-    - antlr4.dfa
-    - antlr4.error
-    - antlr4.tree
-    - antlr4.xpath
-  commands:
-    - pip check
+  number: 1
```
Comment (bollwyvl) [inline recipe/meta.yaml:21]: this can be `noarch: python` #29

Comment (BastianZim) [reply recipe/meta.yaml:21]: Waiting on https://github.com/antlr/antlr4/issues/3350

File: recipe/meta.yaml [hunk 3]
```diff
@@ -12,32 +14,48 @@ source:
...
+
+
+outputs:
+
+  - name: {{ old_name }}
```
Comment (isuruf) [inline recipe/meta.yaml]: Old/deprecated name should be the meta package and the new name should have the actual files in it.

**Comment by BastianZim** [comment]
@bollwyvl Ok so here is what I would do:

1. Close issues and PRs mentioned above on the duplicate feedstock
    1. https://github.com/conda-forge/antlr4-python3-runtime-feedstock/issues/1
    2. https://github.com/conda-forge/antlr4-python3-runtime-feedstock/pull/2
2. Archive duplicate feedstock
3. Merge this PR
4. Create a new v4.9.2 release here with `build: 2`. This is higher than the current build numbers: https://anaconda.org/conda-forge/antlr-python-runtime/files?version=4.9.2
5. Mark the noarch package of the duplicate feedstock as broken.
6. Due to 4., the noarch packages will be replaced with the proper packages from here resulting in (hopefully) no disruption to users.

**Comment by BastianZim** [comment]
@isuruf as some of the above requires core input. Would the approach be acceptable?

File: recipe/0001-Use-entry_points.patch [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+     install_requires=[
+         "typing ; python_version<'3.5'",
+@@ -12,6 +12,10 @@ setup(
+     license='BSD',
+     author='Eric Vergnaud, Terence Parr, Sam Harwell',
+     author_email='eric.vergnaud@wanadoo.fr',
+-    scripts=["bin/pygrun"],
++    entry_points={
++         'console_scripts': [
++             'wsdump=websocket.commands.wsdump:main',
```
Comment (bollwyvl) [inline recipe/0001-Use-entry_points.patch]: should probably terminate in `pygrun=antlr4.commands.pygrun`

**Comment by bollwyvl** [comment]
I'll take a look at this evening, but it won't be immediately...

File: recipe/0001-Use-entry_points.patch (hunk 4 above)
Comment (BastianZim) [reply recipe/0001-Use-entry_points.patch]: Oh shoot sorry forgot to push that.

**Comment by BastianZim** [comment]
Thanks! I'm just not sure how to do this noarch switch but that's nothing urgent.

**Comment by bollwyvl** [comment]
Opened up #31 for another look at it...

**Comment by BastianZim** [comment]
Feel free to also just push to my branch if you want to try something.

**Comment by bollwyvl** [comment]
Merged #31, guess we still have some stuff to do.

**Comment by BastianZim** [comment]
We just need v4.9.2 release with build: 2. The rest has to be done by core. Do you want to create the PR or should I?

**Pull request closed without merging**